### DISCUSSION TITLE
Code snippets for configurations in all Zoxide supported shells installable in Termux, bar Zsh

### DISCUSSION THREAD

**Discussion by b9Joker108** [ajeetdsouza/zoxide#772] [Q&A]
Hi

I use Zsh, with Oh My Zsh!, as a daily driver, in my host environment in Termux. I have come to depend on Zoxide soooooo much. In don't know how I ever did without it. I love it! To all concerned, thank you so much for the code.

So, I am branching out, and exploring all the different shells, installable on Termux. I think I have them all installed, I have lost count, how many there are. I really want to integrate Zoxide into all of their configurations, but I couldn't find any express, instructions, to do so. And, I am a bit of a noob.

I have included the code snippets, that I have in my Zsh config, but I would really appreciate hand-holding, to initiate a solution, for all Zoxide supported shells, installable on Termux. Then, if we could include and actually showcase that information in the Zoxide documentation, that would be wonderful.

Any assistance is greatly appreciated.

```zsh
# Create a temporary file to store fzf's result
export FZF_DEFAULT_OPTS="--print-query --expect=ctrl-m,ctrl-c > /tmp/fzf.result"

# Setting Environment Variables for Zoxide
export _ZO_DATA_HOME="$HOME/.config/zoxide"
export _ZO_ECHO=1
export _ZO_EXCLUDE_DIRS=  #leave this env variable unset
export _ZO_FZF_OPTS=  #leave this env variable unset
export _ZO_MAXAGE=10000
export _ZO_RESOLVE_SYMLINKS=1

# To initialize zoxide, add this to your Zsh config
eval "$(zoxide init zsh)"

# When using zoxide with --no-cmd, alias these internal functions as desired.
#

__zoxide_z_prefix='z#'

# Jump to a directory using only keywords.
function __zoxide_z() {
    # shellcheck disable=SC2199
    if [[ "$#" -eq 0 ]]; then
        __zoxide_cd ~
    elif [[ "$#" -eq 1 ]] && { [[ -d "$1" ]] || [[ "$1" = '-' ]] || [[ "$1" =~ ^[-+][0-9]$ ]]; }; then
        __zoxide_cd "$1"
    elif [[ "$@[-1]" == "${__zoxide_z_prefix}"?* ]]; then
        # shellcheck disable=SC2124
        \builtin local result="${@[-1]}"
        __zoxide_cd "${result:${#__zoxide_z_prefix}}"
    else
        \builtin local result
        # shellcheck disable=SC2312
        result="$(\command zoxide query --exclude "$(__zoxide_pwd)" -- "$@")" &&
            __zoxide_cd "${result}"
    fi
}

# Jump to a directory using interactive search.
function __zoxide_zi() {
    \builtin local result
    result="$(\command zoxide query --interactive -- "$@")" && __zoxide_cd "${result}"
}

# Completions.
if [[ -o zle ]]; then
    function __zoxide_z_complete() {
        # Only show completions when the cursor is at the end of the line.
        # shellcheck disable=SC2154
        [[ "${#words[@]}" -eq "${CURRENT}" ]] || return 0

        if [[ "${#words[@]}" -eq 2 ]]; then
            _files -/
        elif [[ "${words[-1]}" == '' ]] && [[ "${words[-2]}" != "${__zoxide_z_prefix}"?* ]]; then
            \builtin local result
            # shellcheck disable=SC2086,SC2312
            if result="$(\command zoxide query --exclude "$(__zoxide_pwd)" --interactive -- ${words[2,-1]})"; then
                result="${__zoxide_z_prefix}${result}"
                # shellcheck disable=SC2296
                compadd -Q "${(q-)result}"
            fi
            \builtin printf '\e[5n'
        fi
        return 0
    }

    \builtin bindkey '\e[0n' 'reset-prompt'
    [[ "${+functions[compdef]}" -ne 0 ]] && \compdef __zoxide_z_complete __zoxide_z
fi

# As I already have a Zi app, which uses the functions z and zi, I need to create aliases for the Zoxide z and zi functions
alias zz=__zoxide_z
alias zzi=__zoxide_zi
```

**Comment by b9Joker108** [comment]
I now have Zoxide working in Nushell. It was so straightforward, as to be amazing.

Please add tge following to Zoxide documentation:

To add the `zoxide` initialization command to your Nushell configuration, you should save it in your Nushell environment file. You can find the path to this file by running `$nu.env-path` in Nushell. Once you have the path, you can add the `zoxide init nushell` command to the end of your env file using the following command:

```nu
zoxide init nushell | save -f ~/.zoxide.nu
```

After saving the initialization command, you'll also need to source this file in your Nushell configuration file. You can find the path to the config file by running `$nu.config-path` in Nushell. Then, add the following line to the end of your config file:

```shell
source ~/.zoxide.nu
```

This will ensure that `zoxide` is properly initialized each time you start Nushell.

So, two shells down!

**Comment by ajeetdsouza** [comment]
Isn't this information already provided in the README?

https://github.com/ajeetdsouza/zoxide/blob/94d3cba60e24139319c7d370d796967c607b54ef/README.md?plain=1#L279-L302

https://github.com/ajeetdsouza/zoxide/blob/94d3cba60e24139319c7d370d796967c607b54ef/README.md?plain=1#L234-L253
